src/Core/src/Hosting/ImageSources/ImageSourceServiceProvider.cs

```diff
@@ -14,30 +15,35 @@ sealed class ImageSourceServiceProvider : MauiFactory, IImageSourceServiceProvid
 		readonly ConcurrentDictionary<Type, Type> _imageSourceCache = new ConcurrentDictionary<Type, Type>();
 		readonly ConcurrentDictionary<Type, Type> _serviceCache = new ConcurrentDictionary<Type, Type>();
 
+		readonly ImageSourceToImageSourceServiceTypeMapping _imageSourceMapping;
+
 		public ImageSourceServiceProvider(IImageSourceServiceCollection collection, IServiceProvider hostServiceProvider)
 			: base(collection)
 		{
+			_imageSourceMapping = ImageSourceToImageSourceServiceTypeMapping.GetInstance(collection);
 			HostServiceProvider = hostServiceProvider;
 		}
 
 		public IServiceProvider HostServiceProvider { get; }
 
-		public IImageSourceService? GetImageSourceService(Type imageSource) =>
-			(IImageSourceService?)GetService(GetImageSourceServiceType(imageSource));
+		public IImageSourceService? GetImageSourceService(Type imageSource)
+		{
+			var imageSourceService = _serviceCache.GetOrAdd(imageSource, _imageSourceMapping.FindImageSourceServiceType);
+			return (IImageSourceService?)GetService(imageSourceService);
+		}
 
-		public Type GetImageSourceServiceType(Type imageSource) =>
-			_serviceCache.GetOrAdd(imageSource, type =>
+		public Type GetImageSourceServiceType(Type imageSource)
+			=> _serviceCache.GetOrAdd(imageSource, (type) =>
 			{
 				var genericConcreteType = ImageSourceServiceType.MakeGenericType(type);
-
 				if (genericConcreteType != null && GetServiceDescriptor(genericConcreteType) != null)
-					return genericConcreteType;
+					return genericConcreteType!;
 
 				return ImageSourceServiceType.MakeGenericType(GetImageSourceType(type));
 			});
 
-		public Type GetImageSourceType(Type imageSource) =>
-			_imageSourceCache.GetOrAdd(imageSource, CreateImageSourceTypeCacheEntry);
+		public Type GetImageSourceType(Type imageSource)
+			=> _imageSourceCache.GetOrAdd(imageSource, CreateImageSourceTypeCacheEntry);
```

Comment:
Any particular reason to move `=>` to the next line? This deviates from the other `=>` in ImageSourceSerivceProviderExtensions. Is it because of MauiHandlersFactory?

Reply:
There's no reason for that. As I was editing code, I arrived back at the arrow function, but with wrong formatting. I'll revert the change later.